### ISSUE TITLE
Incorporate PHPStan for Static Analysis

### ISSUE THREAD

**Issue by remyperona** [wp-media/wp-rocket#6541]
PHPStan basic configuration has been added in https://github.com/wp-media/wp-rocket/pull/5995.

To go further, as a start we can:
- Add the analysis to the CI lint action
- Start fixing errors at level 0

Once this is done, we can open a new issue to fix level 1 and go on like this.
